quarkus/extensions/kogito-quarkus-serverless-workflow-extension/kogito-quarkus-serverless-workflow-integration-test/src/test/java/org/kie/kogito/quarkus/workflows/TokenPropagationIT.java

```diff
@@ -57,10 +66,76 @@ void tokenPropagations() {
+        //Thread.sleep(20000);
+        waitForProcessCompletion(processInstanceId, Duration.ofSeconds(25));
+        validateExternalServiceInvocations();
     }
 
     protected static String buildProcessInput(String query) {
         return "{\"workflowdata\": {\"query\": \"" + query + "\"} }";
     }
+
+    private void waitForProcessCompletion(String processInstanceId, Duration timeout) {
+        long startTime = System.currentTimeMillis();
+        long timeoutMs = timeout.toMillis();
+
+        while (System.currentTimeMillis() - startTime < timeoutMs) {
+            try {
+                // Check if process still exists - 404 means it completed and was cleaned up
+                int statusCode = given()
+                        .contentType("application/json")
+                        .accept("application/json")
+                        .when()
+                        .get("/token_propagation/" + processInstanceId)
+                        .then()
+                        .extract()
+                        .statusCode();
+
+                if (statusCode == 404) {
+                    LOGGER.info("Process instance {} completed successfully (404 - cleaned up)", processInstanceId);
+                    return;
+                }
+
+                Thread.sleep(1000); // Wait 1 second before checking again
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt();
+                throw new RuntimeException("Interrupted while waiting for process completion", e);
+            } catch (Exception e) {
+                LOGGER.debug("Error checking process state (will retry): {}", e.getMessage());
+                try {
+                    Thread.sleep(1000);
+                } catch (InterruptedException ie) {
+                    Thread.currentThread().interrupt();
+                    throw new RuntimeException("Interrupted while waiting for process completion", ie);
+                }
+            }
+        }
+
+        throw new RuntimeException("Process instance " + processInstanceId + " did not complete within " + timeout);
+    }
+
+    private void validateExternalServiceInvocations() {
+        WireMockServer wm = TokenPropagationExternalServicesMock.getInstance();
+
+        assertCalledExactlyWithAuth(wm, "/token-propagation-external-service1/executeQuery1", "Bearer " + AUTHORIZATION_TOKEN, 2);
+        assertCalledExactlyWithAuth(wm, "/token-propagation-external-service2/executeQuery2", "Bearer " + AUTHORIZATION_TOKEN, 2);
+        assertCalledExactlyWithAuth(wm, "/token-propagation-external-service3/executeQuery3", "Bearer " + SERVICE3_AUTHORIZATION_TOKEN, 2);
+        assertCalledExactlyWithAuth(wm, "/token-propagation-external-service4/executeQuery4", "Bearer " + SERVICE4_AUTHORIZATION_TOKEN, 2);
+        assertCalledExactlyWithAuth(wm, "/token-propagation-external-service5/executeQuery5", "Bearer " + KeycloakServiceMock.KEYCLOAK_ACCESS_TOKEN, 2);
```

Comment:
As only token expectation is changing for each call, probably it would be easier to maintain if we group it in a list to iterate over it:

```suggestion
List<TokenExpectation> expectations = List.of(
    new TokenExpectation("/service1", "Bearer " + AUTHORIZATION_TOKEN),
    new TokenExpectation("/service2", "Bearer " + AUTHORIZATION_TOKEN),
    new TokenExpectation("/service3", "Bearer " + SERVICE3_AUTHORIZATION_TOKEN),
    new TokenExpectation("/service4", "Bearer " + SERVICE4_AUTHORIZATION_TOKEN),
    new TokenExpectation("/service5", "Bearer " + KeycloakServiceMock.KEYCLOAK_ACCESS_TOKEN)
);

expectations.forEach(e -> assertCalledExactlyWithAuth(wm, e.url(), e.token(), 2));
```